view/src/storage.rs

```diff
@@ -942,6 +942,8 @@ impl Storage {
             return Ok(BTreeMap::new());
         }
 
+        // This query gives advice about notes which are known but which have not already been recorded as spendable,
+        // in part to avoid revealing information about which notes have been spent.
```

Comment:
I don't think that's the reason -- we're already inside the trust boundary, the reason we want to exclude notes that have been marked as spendable is that we only want to give info on notes that have not been scanned yet: 
```
    /// Given a list of note commitments, this method checks whether any of them
    /// correspond to notes that have been recorded in the database but not yet
    /// observed during scanning.
```

Two points:

1. Do we remember why this was important?
2. I think this means the TxP code from earlier this week that uses `scan_advice` is incorrect and won't give the necessary information.

Reply:
1. Not personally, I was trying to divine the reasoning from the comment in #1736 but didn't remember context
2. Oh hmm, what will we need to do to fix that?

Reply:
Looks like this was the reason: https://github.com/penumbra-zone/penumbra/pull/1736#discussion_r1045175372

For (2), instead of using scan_advice for TxPs, we should just query notes by commitment.